### PR TITLE
[Security/Http] Ignore invalid URLs found in failure/success paths

### LINKED ISSUE #43567: [Security] Do not throw exception for a non-existing route defined in `_target_path` when login
**Symfony version(s) affected**: 5.3.x and I guess 4.4.x

**Description**

When [using `_target_path` in a login](https://symfony.com/doc/current/security/form_login.html#control-the-redirect-using-request-parameters), if the value is a non-existing route it throws a `Symfony\Component\Routing\Exception\
RouteNotFoundException`.

**How to reproduce**
In a login form allowing `_target_path`, modify the value to send a value which is not a route and neither starts with `http`.

**Possible Solution**
I guess if the value is not starting with `http`, it could be checked if it's an existing route and if not set the `$path` to `/` in

https://github.com/symfony/symfony/blob/732acf56fd6e0a6a3af57595a724728b959a5cbf/src/Symfony/Component/Security/Http/HttpUtils.php#L61-L70

or catch the `RouteNotFoundException` and redirect to `/` maybe.

**Additional context**

image

I came across this when someone tried to set `_target_path` to `<script>something</script>` (which does nothing because it's just used as a key from an array).

### PR DESCRIPTION
**Pull request by nicolas-grekas**
| Q             | A
| ------------- | ---
| Branch?       | 4.4
| Bug fix?      | yes
| New feature?  | no
| Deprecations? | no
| Tickets       | Fix #43567
| License       | MIT
| Doc PR        | -

### CODE REVIEW

**Review by maxhelias — APPROVED** [review]

**Review by chalasr — APPROVED** [review]

**Review by wouterj — APPROVED** [review]

File: src/Symfony/Component/Security/Http/Authentication/DefaultAuthenticationFailureHandler.php [hunk 1]
```diff
@@ -70,31 +70,34 @@ public function setOptions(array $options)
...
-        }
+        $options = $this->options;
+        $failureUrl = ParameterBagUtils::getRequestParameterValue($request, $options['failure_path_parameter']);
 
-        if (null === $this->options['failure_path']) {
-            $this->options['failure_path'] = $this->options['login_path'];
+        if (\is_string($failureUrl) && str_starts_with($failureUrl, '/')) {
+            $options['failure_path'] = $failureUrl;
+        } elseif ($this->logger && $failureUrl) {
+            $this->logger->debug(sprintf('Ignoring query parameter "%s": not a valid URL.', $options['failure_path_parameter']));
```
Comment (wouterj) [inline src/Symfony/Component/Security/Http/Authentication/DefaultAuthenticationFailureHandler.php:79]: Should we include the value in the log message/context?

File: src/Symfony/Component/Security/Http/Authentication/DefaultAuthenticationFailureHandler.php [hunk 2]
```diff
@@ -70,31 +70,34 @@ public function setOptions(array $options)
...
         }
 
-        if ($this->options['failure_forward']) {
+        $options['failure_path'] ?? $options['failure_path'] = $options['login_path'];
```
Comment (wouterj) [inline src/Symfony/Component/Security/Http/Authentication/DefaultAuthenticationFailureHandler.php:82]: I must confess that I'm not at all sure if I understand what is going on here. Is this a leftover?

**Review by wouterj — CHANGES_REQUESTED** [review]
sorry, too quick

File: src/Symfony/Component/Security/Http/Authentication/DefaultAuthenticationFailureHandler.php (hunk 2 above)
Comment (nicolas-grekas) [reply src/Symfony/Component/Security/Http/Authentication/DefaultAuthenticationFailureHandler.php:82]: This is `$options['failure_path'] ??= $options['login_path'];` written for PHP 7.1+

File: src/Symfony/Component/Security/Http/Authentication/DefaultAuthenticationFailureHandler.php (hunk 1 above)
Comment (nicolas-grekas) [reply src/Symfony/Component/Security/Http/Authentication/DefaultAuthenticationFailureHandler.php:79]: I don't think so: this is user input, might be just garbage. The query parameter name is enough to me.

**Review by fabpot — APPROVED** [review]

**Comment by fabpot** [comment]
Thank you @nicolas-grekas.

**Comment by crmpicco** [comment]
Is there a recommended way to pass through the `target_path` now? It appears this change has broken the login functionality in some of my code. I created [a discussion](https://github.com/symfony/symfony/discussions/46677) relating to it last week.

**Pull request merged**